cms/djangoapps/contentstore/views/course.py

```diff
@@ -1317,6 +1331,26 @@ def update_usage_info(store, course, configuration):
         configuration_json['usage'] = usage_information.get(configuration.id, [])
         return configuration_json
 
+    @staticmethod
+    def get_or_create_content_group_configuration(course):
```

Comment:
Yeah, I found this class pretty confusing when I first came to it. Though it does resemble the solution you describe, right?